classes/plugins/PluginRegistry.inc.php

```diff
@@ -258,9 +249,12 @@ public static function getCategories()
      */
     public static function loadAllPlugins($enabledOnly = false)
     {
-        // Retrieve and register categories (order is significant).
-        foreach (self::getCategories() as $category) {
-            self::loadCategory($category, $enabledOnly);
+        static $isLoaded;
```

Comment:
I doubt this is going to be called many times, but in case it does, it should run faster.